x/gov/keeper/proposal.go

```diff
@@ -146,18 +155,22 @@ func (keeper Keeper) CancelProposal(ctx context.Context, proposalID uint64, prop
 		return types.ErrInvalidProposal.Wrap("proposal should be in the deposit or voting period")
 	}
 
-	// Check proposal voting period is ended.
-	if proposal.VotingEndTime != nil && proposal.VotingEndTime.Before(sdkCtx.HeaderInfo().Time) {
-		return types.ErrVotingPeriodEnded.Wrapf("voting period is already ended for this proposal %d", proposalID)
+	// Check proposal is not too far in voting period to be canceled
+	if proposal.VotingEndTime != nil {
+		currentTime := sdkCtx.HeaderInfo().Time
+
+		maxCancelPeriodRate := sdkmath.LegacyMustNewDecFromStr(params.ProposalCancelMaxPeriod)
```

Comment:
nit: what does rate mean in `maxCancelPeriodRate`? Maybe `maxCancelPeriod` or `maxCancelPeriodDuration` and rename `maxCancelPeriod` to `maxCancelTime`?

Reply:
nvm, it's not a duration, my bad